contents/code/tilingmanager.js

```diff
@@ -575,6 +582,37 @@ function TilingManager(timer) {
                                       print(err, "in i3-layout-set-normal-mode");
                                   }
                               });
+        KWin.registerShortcut("TILING: Move Window To New Desktop",
+                              "Move Window To New Desktop",
+                              "Meta+Shift+D",
+                              function() {
+                                  try {
+                                      var client = workspace.activeClient;
+                                      if (client != null) {
```

Comment:
I know we do this kind of `if` all over the place, but I've come to like guard-ifs more for this sort of thing:

```js
if (client == null) return;
```

Saves a level of indentation.

Reply:
OK, I can change it. I don't really like this sort of returns all over the place. TBH, it is even prohibited by our coding style at the company I currently work for.

Reply:
Not "all over the place", but just at the beginning of the function to check a precondition or two.

I know that many prohibit multiple-returns in general out of a fear of bad control-flow, but I feel like this makes it _more_ obvious, not less, especially once you've gotten used to it.

The problem is that the `if (!thing) {` block can be quite long - it's the entire function here! - so you lose track of what that `}` at the end was for, or why you're nested so deep.

Reply:
>I know that many prohibit multiple-returns in general out of a fear of bad control-flow, but I feel like this makes it more obvious, not less, especially once you've gotten used to it.

We don't allow it here because of the tools we use for code coverage analysis for certification purposes.

>The problem is that the if (!thing) { block can be quite long - it's the entire function here!

So, what's your decision here? Your project - your rules ;)

Reply:
>So, what's your decision here? Your project - your rules ;)

My rules are that there are no rules.

Well, not quite, but let's not waste them on micromanaging code style.